AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ExtraHandler.java

```diff
@@ -72,12 +89,34 @@ public void process(Element element, HasExtras holder) {
 		}
 
 		JFieldVar extraKeyStaticField = createStaticExtraField(holder, extraKey, fieldName);
-		injectExtraInComponent(element, holder, extraKeyStaticField, fieldName);
-
 		if (holder instanceof HasIntentBuilder) {
 			String docComment = getProcessingEnvironment().getElementUtils().getDocComment(element);
-			createIntentInjectionMethod(element, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
+			createIntentInjectionMethod(param, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
```

Comment:
also this would duplicate some of the code already in `InjectHelper` that is used to manage the injection with that method